govc/namespace/ls.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright (c) 2024-2024 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package namespace
+
+import (
+	"context"
+	"flag"
+	"io"
+
+	"github.com/vmware/govmomi/vapi/namespace"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+)
+
+type lsResult []namespace.NamespacesInstanceSummary
+
+func (r lsResult) Write(w io.Writer) error {
+	return nil
+}
+
+type ls struct {
+	*flags.ClientFlag
+	*flags.OutputFlag
+}
+
+func init() {
+	cli.Register("namespace.ls", &ls{})
+}
+
+func (cmd *ls) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.ClientFlag, ctx = flags.NewClientFlag(ctx)
+	cmd.ClientFlag.Register(ctx, f)
+
+	cmd.OutputFlag, ctx = flags.NewOutputFlag(ctx)
+}
+
+func (cmd *ls) Process(ctx context.Context) error {
+	if err := cmd.ClientFlag.Process(ctx); err != nil {
+		return err
+	}
+	if err := cmd.OutputFlag.Process(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cmd *ls) Description() string {
+	return `Displays the list of vSphere Namespaces. 
+
+Examples:
+  govc namespace.ls`
+}
+
+func (cmd *ls) Run(ctx context.Context, f *flag.FlagSet) error {
+	rc, err := cmd.RestClient()
+
+	if err != nil {
+		return err
+	}
+
+	nm := namespace.NewManager(rc)
+
+	if d, err := nm.ListNamespaces(ctx); err != nil {
+		return err
+	} else {
```

Comment:
nit: no need for `else` since you returned earlier (here and other places too)
```
if d, err := nm.ListNamespaces(ctx); err != nil {
	return err
}

// Not too sure what this thing is doing but suggest:
cmd.JSON = !cmd.All()
[...]
```

Reply:
You're right, it's better with less blocks
I fixed all occurrences of this pattern.